machine-learning/sagemaker-edge-manager/create_components.py

```diff
@@ -0,0 +1,237 @@
+        response = greengrass_client.list_component_versions(
+            arn="arn:aws:greengrass:{}:{}:components:{}".format(region, account, c_name),
+        )
+        component_versions = response["componentVersions"]
+        if len(component_versions) > 0:
+            versions = component_versions[0]["componentVersion"]
+            split = versions.split("-")[0].split(".")
+            major = split[0]
+            minor = split[1]
+            patch = split[2]
+            return "{}.{}.{}".format(major, minor, str(int(patch) + 1))
+        else:
+            return c_version
+    except Exception as e:
+        print("Error getting the latest version of the component.\nException: {}".format(e))
+
+
+def create_bucket():
+    if bucket_exists(bucket):
+        print("Bucket:{} already exists.".format(bucket))
+        return
+    try:
+        if region is None or region == "us-east-1":
+            s3_client.create_bucket(Bucket=bucket)
+        else:
+            location = {"LocationConstraint": region}
+            s3_client.create_bucket(Bucket=bucket, CreateBucketConfiguration=location)
+    except Exception as e:
+        print("Failed to create the bucket: {}.\nException: {}".format(bucket, e))
+        exit(1)
+    print("Successfully created the artifacts bucket {}".format(bucket))
+
+
+def bucket_exists(bucket_name):
+    response = s3_client.list_buckets()
+    for bucket in response["Buckets"]:
+        if bucket["Name"] == bucket_name:
+            return True
+    return False
+
+
+def upload_artifacts(file_name, object_name=None):
+    if object_name is None:
+        object_name = file_name
+    try:
+        response = s3_client.upload_file(file_name, bucket, object_name)
+    except Exception as e:
+        print(
+            "Failed to upload the artifacts to the bucket {} with key {}.\nException: {}".format(
+                bucket, object_name, e
+            )
+        )
+        exit(1)
+    print(
+        "Successfully uploaded the artifacts to the bucket {} with key {}.".format(
+            bucket, object_name
+        )
+    )
+
+
+def create_components():
+    for component_recipe in os.listdir(build_recipes_path):
+        recipe_file_path = os.path.join(build_recipes_path, component_recipe)
+        with open(
+            recipe_file_path,
+        ) as f:
+            try:
+                response = greengrass_client.create_component_version(inlineRecipe=f.read())
+            except Exception as e:
+                print(
+                    "Failed to create the component using the recipe at {}.\nException: {}".format(
+                        recipe_file_path, e
+                    )
+                )
+                exit(1)
+
+
+def copy_sample_models(path, model_names):
+    try:
+        response = s3_client.head_bucket(Bucket=public_models_bucket)
+        if response["ResponseMetadata"]["HTTPStatusCode"] == 200:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Your code is attempting to access the response metadata, `ResponseMetadata`, without performing a `None` check on the response object. This might cause a NoneType TypeError. To prevent this, we recommend that you add a `None` check on the response object before accessing the response metadata.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html)